### PR TITLE
Add Catch2

### PR DESCRIPTION
**Pull request by ncihnegn**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/catch2.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class Catch2 < Formula
+  desc "Modern, C++-native, header-only, test framework"
+  homepage "https://github.com/catchorg/Catch2"
+  url "https://github.com/catchorg/Catch2/archive/v2.5.0.tar.gz"
+  sha256 "720c84d18f4dc9eb23379941df2054e7bcd5ff9c215e4d620f8533a130d128ae"
+
+  depends_on "cmake" => :build
+
+  def install
+    inreplace "CMakeLists.txt", "{CMAKE_INSTALL_LIBDIR}/cmake", "{CMAKE_INSTALL_DATAROOTDIR}/cmake"
```
Comment (fxcoudert) [inline Formula/catch2.rb]: Why is this necessary? Has it been reported upstream?

File: Formula/catch2.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    system "cmake", "-Bbuild", "-H.", "-DBUILD_TESTING=OFF", *std_cmake_args
```
Comment (fxcoudert) [inline Formula/catch2.rb]: `-H` is not a documented `cmake` option, so we shouldn't be using it.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/catch2.rb (hunk 2 above)
Comment (ncihnegn) [reply Formula/catch2.rb]: Agree.

File: Formula/catch2.rb (hunk 1 above)
Comment (ncihnegn) [reply Formula/catch2.rb]: No, this is not really necessary. It works out of box so nothing to report. But it feels better if we use the default path.
